functions/src/lobby.ts

```diff
@@ -185,3 +193,74 @@ export const collectPromptAnswers = functions.firestore
       // TODO #32 create a one on one chat and store the promptAnswers in it
     });
   });
+
+export async function addRole(lobbyDocRef: firestore.DocumentReference<Lobby>) {
+  const lobby = await lobbyDocRef.get();
+  if (lobby.exists === false) {
+    return { error: "Lobby doesn't exist!" };
+  }
+  const privatePlayerCollection = getPrivatePlayerCollection(lobbyDocRef);
+  const privatePlayers = (await privatePlayerCollection.get()).docs;
+
+  const { players, uids } = lobby.data() as Lobby;
+
+  for (let i = 0; i < players.length; i++) {
+    //  private player id = user uid
+    const privateId = privatePlayers[i].id;
+    // private player info
+    const privateInfo = privatePlayers[i].data();
+    // get player
+    const player = players[uids.indexOf(privateId)];
+    // replace the role of the player
+    player.role = privateInfo.role;
+  }
+  let catCount = 0;
+  let catfishCount = 0;
+  const catfishDisplayname: string[] = [];
+  for (let i = 0; i < uids.length; i++) {
+    // check the alive players
+    if (players[i].alive == true) {
+      if (players[i].role == "CAT") {
+        catCount++;
+        // within the alive players, count the number that are cats
+      } else {
+        let temp = "";
+        temp = players[i].displayName;
+        catfishCount++;
+        catfishDisplayname.push(temp);
+      }
+    }
+    // check any catfish who are not alive and add their display name to the list
+    else {
+      if (players[i].role == "CATFISH") {
+        let temp = "";
+        temp = players[i].displayName;
+        catfishDisplayname.push(temp);
+      }
+    }
+  }
+
+  let winner: Role | undefined;
+  if (catfishCount == 0) {
+    winner = "CAT";
+  }
+  if (catCount <= catfishCount) {
+    winner = "CATFISH";
+  }
```

Comment:
these do not cover every case

Reply:
I'm honestly struggling to figure out what case I'm missing. `if(catCount == 0)`?

Reply:
For example if there is 1 catfish and 3 cats.

I feel like this will be more obvious once you integrate with the voting code

Reply:
If there’s 1 catfish and 3 cats the game would continue right? I think neither role wins yet in that situation